### PR TITLE
3885/update workflow completely before refreshing checker

### PR DESCRIPTION
**Pull request by NatalieEO**
#3885
Not entirely sure what was going on, but this bug was only for the workflow mentioned in the ticket. Seems like you need to reset the workflow's default version before trying to refresh its checker workflow.

### CODE REVIEW

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 1]
```diff
@@ -4924,7 +4924,7 @@ paths:
   /organizations/{organizationId}/collections/{collectionId}:
     get:
       description: Retrieve a collection by ID. Supports optional authentication.
-      operationId: getCollectionById_1
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml:4927]: :sob:

**Review by denis-yuen — APPROVED** [review]

**Review by coverbeck — APPROVED** [review]
I think I see what is going on; on line 508 of WorkflowResource, it calls refresh for the checker workflow, which then on line 422 calls `EntryVersionHelper.removeSourceFilesFromEntry(workflow, sessionFactory)` -- that method does a `session.flush()`, which commits to the DB. At this point the default is  pointing to the deleted version, so you get the error. And your PR fixes it because, as you said, you've set the default before it tries to persist.

I suggest a comment in the code about why it's important to preserve the new order.

Also, I don't like calling the `refresh` endpoint like that; I'm not clear what impact the annotations have. I always try to avoid it. But beyond the scope of this PR.

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 2]
```diff
@@ -5556,7 +5556,7 @@ paths:
   /organizations/{organizationName}/collections/{collectionName}/name:
     get:
       description: Retrieve a collection by name. Supports optional authentication.
-      operationId: getCollectionById
+      operationId: getCollectionById_1
```
[the remark this answers is not in the document]
Comment (NatalieEO) [reply dockstore-webservice/src/main/resources/openapi3/openapi.yaml:5559]: Hm, I built the UI locally based off the circleCI jar and seems like the number got ignored.

[the remark this answers is not in the document]
Comment (coverbeck) [reply dockstore-webservice/src/main/resources/openapi3/openapi.yaml:5559]: 1. The UI seems to be invoking the Swagger-generated client, so it's not running into this
2. I see OrganizationsService.getCollectionById1 in my OpenAPI generated code, so if anybody is using this, it probably will break. But I don't think anybody is. Might be worth fixing. I think you need to just change the operation id for the name one (it's a duplicate), and we can let that one break.

**Review by garyluu — APPROVED** [review]

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 3]
```diff
@@ -5556,7 +5556,7 @@ paths:
   /organizations/{organizationName}/collections/{collectionName}/name:
```
[the remark this answers is not in the document]
Comment (garyluu) [reply dockstore-webservice/src/main/resources/openapi3/openapi.yaml:5556]: Ya, a lot of Dockstore endpoints aren't RESTful. oh well.

**Pull request merged**